akka-actor/src/main/scala/akka/dispatch/affinity/AffinityPool.scala

```diff
@@ -0,0 +1,341 @@
+/**
+ *  Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.dispatch.affinity
+
+import java.util
+import java.util.Collections
+import java.util.concurrent._
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.locks.{ Lock, LockSupport, ReentrantLock }
+import akka.dispatch._
+import akka.dispatch.affinity.AffinityPool._
+import com.typesafe.config.Config
+import net.openhft.affinity.{ AffinityStrategies, AffinityThreadFactory }
+
+import scala.annotation.tailrec
+import scala.collection.JavaConversions._
+import scala.collection.mutable
+
+class AffinityPool(parallelism: Int, affinityGroupSize: Int, tf: ThreadFactory, waitingStrategy: WaitingStrategy) extends AbstractExecutorService {
+
+  if (parallelism <= 0)
+    throw new IllegalArgumentException("Size of pool cannot be less or equal to 0")
+
+  // Held while starting/shutting down workers/pool in order to make
+  // the operations linear and enforce atomicity. An example of that would be
+  // adding a worker. We want the creation of the worker, addition
+  // to the set and starting to worker to be an atomic action. Using
+  // a concurrent set would not give us that
+  private val bookKeepingLock = new ReentrantLock()
```

Comment:
If this thing doesn't need any external dependnecies then it could probably go into `akka-actor`, right?